packages/cross-chain/src/services/IntentTracker.ts

```diff
@@ -191,6 +209,53 @@ export class IntentTracker {
         }
     }
 
+    /**
+     * Start tracking an intent with event-based updates
+     * Uses watchIntent() internally and emits events for each status change
+     * Returns a promise that resolves when the intent is filled or expired
+     *
+     * @param params - Watch parameters
+     * @returns Promise that resolves with final intent status
+     *
+     * @example
+     * ```typescript
+     * const tracker = new IntentTracker(...);
+     *
+     * tracker.on('update', (update) => console.log(update.status));
+     * tracker.on('filled', (update) => console.log('Filled!', update.fillTxHash));
+     *
+     * const finalStatus = await tracker.startTracking({
+     *   txHash,
+     *   originChainId,
+     *   destinationChainId
+     * });
+     * ```
+     */
+    async startTracking(params: WatchIntentParams): Promise<IntentStatusInfo> {
+        try {
+            for await (const update of this.watchIntent(params)) {
+                // Emit generic update event
+                this.emit("update", update);
+
+                // Emit status-specific event
+                this.emit(update.status, update);
+
+                // Stop if we reached a final state
+                if (update.status === "filled" || update.status === "expired") {
+                    break;
+                }
```

Comment:
Should we add a timeout option here? Just in case the order doesnt expire for some reason